.travis.yml

```diff
@@ -26,15 +31,43 @@ matrix:
       env: Protocol=Mongo Backend=MySQL
     - jdk: oraclejdk7
       env: Protocol=MongoReplSet Backend=Postgres
+    - jdk: oraclejdk7
+      env: DOCKER_TEST=TRUE
   allow_failures:
     - env: Protocol=MongoReplSet Backend=Postgres
 
+before_install:
+  -  if [[ "$DOCKER_TEST" == "TRUE" ]]; then sudo /etc/init.d/postgresql stop; fi
+
+install:
+  -  if [[ "$DOCKER_TEST" == "TRUE" ]]; then sudo sh -c 'echo "deb https://apt.dockerproject.org/repo ubuntu-precise main" > /etc/apt/sources.list.d/docker.list'; fi
```

Comment:
I would group some of this to don't repeat that much `[[ "$DOCKER_TEST" == "TRUE" ]]`.

Perhaps you could move all this to a `setup_docker_env` function, `source` it and use it in just one line? I think it could be more readable that way.